lib/matplotlib/axes/_axes.py

```diff
@@ -2420,22 +2423,38 @@ def stem(self, *args, **kwargs):
         except IndexError:
             basefmt = kwargs.pop('basefmt', 'r-')
 
+        # Check if the user wants a vertical stem plot
+        vertical = kwargs.pop('vertical', None)
```

Comment:
might as well have pop default to `False`

This is one function that would greatly benefit from dropping python 2 as we could write the signature as

```python
def stem(*args, vertical=False, bottom=None, label=None, **kwargs):
```